pallets/compliance-manager/src/lib.rs

```diff
@@ -765,6 +832,71 @@ impl<T: Trait> Module<T> {
         }
         Err(Error::<T>::ComplianceRequirementTooComplex.into())
     }
+
+    // Know whether sender and receiver has valid scope claim or not before checking the transfer conditions.
+    fn is_sender_and_receiver_has_valid_scope_claim(
+        ticker: &Ticker,
+        from_did_opt: Option<IdentityId>,
+        to_did_opt: Option<IdentityId>,
+        primary_issuance_agent: Option<IdentityId>,
+    ) -> bool {
```

Comment:
I'd suggest refactoring the whole function like so:
```rust
        let has_scope_claim = |default, did_opt| did_opt.map_or(default, |did| {
            let ctype = ConditionType::HasValidProofOfInvestor(*ticker);
            let condition = &Condition::new(ctype, vec![did]);
            Self::is_condition_satisfied(ticker, did, condition, primary_issuance_agent);
        });
        has_scope_claim(false, to_did_opt) && has_scope_claim(true, from_did_opt)
```
Simpler & highlights the actual difference in to/from.